catroid/src/main/java/org/catrobat/catroid/io/ProjectAndSceneScreenshotLoader.java

```diff
@@ -115,6 +120,41 @@ public File getScreenshotFile(String projectName, String sceneName, boolean isBa
 		return screenshotLoader.getScreenshotFile();
 	}
 
+	public String getScreenshotScene(File projectDir) {
+		FilenameFilter filter = new FilenameFilter() {
+
+			public boolean accept(File f, String name) {
+				return name.endsWith("screenshot.png");
+			}
+		};
+		File[] projectScreenshots = projectDir.listFiles(filter);
+		if (projectScreenshots == null || projectScreenshots.length != 0) {
+			return null;
+		}
```

Comment:
I implemented a new way screenshots are handled. Before we always took the screenshot of the default scene, if there is none, the project thumbnail is empty. From now on we want to use the most recent manual screenshot if available, else the most recent automatic screenshot. Everytime a screenshot is saved, the current screenshot in the project folder is overwritten, so we always have the most recent screenshot.
In this function I'm searching for the Scene thats has the most recent screenshot. (because the screenshot loader has the scenename as parameter). If the projectfolder has a screenshot, I just return null, because we do not need the scenename.
Here I check if the screenshot is available, if I find a screenshot in the project folder i don't need a scenename.